devito/data/data.py

```diff
@@ -170,6 +170,9 @@ def _is_mpi_distributed(self):
     def __repr__(self):
         return super(Data, self._local).__repr__()
 
+    def __str__(self):
+        return super(Data, self._local).__str__()
```

Comment:
nitpicking: `__str__ = __repr__` works to

Reply:
But this will result in slightly different outputs right? So with `__str__ = __repr__`, `print(f.data[some_slice])` will result in `Data([some_array], dtype=float32)` being printed. Where as with the current code you'd get only `[some_array]` (which mimics the numpy array behavior).

Reply:
Rhodri is right. I was tricked too for a second like Mathias; note that `__repr__` is calling `.__repr__()`, while `__str__`  is calling `.__str__()`

Reply:
Oh right my bad, always bit confused but the difference. GTG